File: server/src/main/java/sk/gdpr/gdpranonymizer/model/dto/GdprSearchTreeRequest.java
package sk.gdpr.gdpranonymizer.model.dto;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class GdprSearchTreeRequest {
    @NotNull
    private Long dataSourceId;  // node root id
    @NotNull
    private List<GdprDataSearchCriteria> dataTypes; // gdpr data of the root with search value
}

File: web/src/app/anonymization/model/gdpr-search-tree-request.ts
import {ColumnType} from '../../model/column-type';

export interface GdprSearchTreeRequest {
  dataSourceId: number;                 // ID of root node in configuration
  dataTypes: GdprDataSearchCriteria[];  // search criteria
}


export interface GdprDataSearchCriteria {
  id: number;
  name: string;
  columnType: ColumnType;
  value: any;
}
